terraform/examples/offline.tf

```diff
@@ -0,0 +1,761 @@
+  description = "hosts that are allowed to speak to the private ports of the ephemeral services."
+  vpc_id      = module.vpc.vpc_id
+
+  # cassandra non-TLS
+  egress {
+    from_port   = 7000
+    to_port     = 7000
+    protocol    = "tcp"
+    cidr_blocks = ["172.17.0.0/20"]
+  }
+
+  # cassandra TLS
+  egress {
+    from_port   = 9160
+    to_port     = 9160
+    protocol    = "tcp"
+    cidr_blocks = ["172.17.0.0/20"]
+  }
+
+  # ElasticSearch
+  egress {
+    from_port   = 9300
+    to_port     = 9300
+    protocol    = "tcp"
+    cidr_blocks = ["172.17.0.0/20"]
+  }
+
+  # minio
+  egress {
+    from_port   = 9000
+    to_port     = 9000
+    protocol    = "tcp"
+    cidr_blocks = ["172.17.0.0/20"]
+  }
+
+  # minio
+  egress {
+    from_port   = 9092
+    to_port     = 9092
+    protocol    = "tcp"
+    cidr_blocks = ["172.17.0.0/20"]
+  }  
+
+  tags = {
+    Name = "ephemeral_private"
+  }
+}
+
+# A security group for ephemeral service nodes. should be added to them only.
+resource "aws_security_group" "ephemeral_node" {
+  name        = "ephemeral_node"
+  description = "hosts that host ephemeral services."
+  vpc_id      = module.vpc.vpc_id
+
+  # incoming cassandra clients
+  ingress {
+    from_port   = 9042
+    to_port     = 9042
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.talk_to_ephemeral.id}"]
+  }
+
+  # incoming elasticsearch clients.
+  ingress {
+    from_port   = 9200
+    to_port     = 9200
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.talk_to_ephemeral.id}"]
+  }
+
+  # incoming minio clients.
+  ingress {
+    from_port   = 9000
+    to_port     = 9000
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.talk_to_ephemeral.id}"]
+  }
+
+  # incoming minio clients.
+  ingress {
+    from_port   = 9092
+    to_port     = 9092
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.talk_to_ephemeral.id}"]
+  }
+
+  # other cassandra nodes (non-TLS)
+  ingress {
+    from_port   = 7000
+    to_port     = 7000
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.ephemeral_private.id}"]
+  }
+
+  # other cassandra nodes (TLS)
+  ingress {
+    from_port   = 9160
+    to_port     = 9160
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.ephemeral_private.id}"]
+  }
+
+  # other elasticsearch nodes
+  ingress {
+    from_port   = 9300
+    to_port     = 9300
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.ephemeral_private.id}"]
+  }
+
+  # other minio nodes
+  ingress {
+    from_port   = 9000
+    to_port     = 9000
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.ephemeral_private.id}"]
+  }
+
+  # other minio nodes
+  ingress {
+    from_port   = 9092
+    to_port     = 9092
+    protocol    = "tcp"
+    security_groups = ["${aws_security_group.ephemeral_private.id}"]
+  }
+
+  tags = {
+    Name = "ephemeral_node"
+  }
+}
```

Comment:
I'm not exactly sure what to do about security groups. I suppose we can, for the time being, keep these hardcoded ports as-is, and move all the `aws_security_group` blocks into `terraform/modules/aws_security_groups/main.tf`